src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs

```diff
@@ -164,11 +164,12 @@ private T GetSpecializedCollection<T>(int slot, Func<HttpRequestHeaders, T> crea
         /// <value>The value of the <see langword=":protocol" /> pseudo-header for an HTTP request.</value>
         public string? Protocol
         {
-            get => _protocol;
+            get => _specialCollectionsSlots is null ? null : (string?)_specialCollectionsSlots[ProtocolSlot];
```

Comment:
This also means that if we get an expect or protocol header and we don't get any of the other "special" ones, we allocate the ~100 byte array when we wouldn't have in the past.  Outside of YARP, we expect this is the right tradeoff?

Reply:
Regarding `Expect`, yes, it should be uncommon enough. In fact you've already made that tradeoff in the past https://github.com/dotnet/runtime/commit/7273e9def81556ffa6649394a27d6585d8ccb703, but the change was later reverted for Expect (108bf6bbf767c028d3714ad434e1d7c8e2068eac) as we were accessing the property ourselves. This is no longer the case as of #63057.

Regarding `Protocol`, that is the field we just added to support HTTP/2 WebSockets. I would expect those to be rare compared to all other requests.